src/workerd/api/actor-state.h

```diff
@@ -348,11 +350,41 @@ class DurableObjectState: public jsg::Object {
   jsg::Promise<jsg::Value> blockConcurrencyWhile(jsg::Lock& js,
       jsg::Function<jsg::Promise<jsg::Value>()> callback);
 
+  void acceptWebSocket(jsg::Ref<WebSocket> ws, kj::Array<kj::String> tags);
+  // TODO(now): We get a bunch of errors when using an Array of StringPtr's -- figure this out.
+  //
+  // Adds a WebSocket to the set attached to this object.
+  // `ws.accept()` must NOT have been called separately.
+  // Once called, any incoming messages will be delivered
+  // by calling the Durable Object's webSocketMessage()
+  // handler, and webSocketClose() will be invoked upon
+  // disconnect.
+  //
+  // After calling this, the WebSocket is accepted, so
+  // its send() and close() methods can be used to send
+  // messages, but its addEventListener() method won't
+  // ever receive any events as they'll be delivered to
+  // the DurableObject instead.
+  //
+  // `tags` are string tags which can be used to look up
+  // the WebSocket with getWebSockets().
+
+  kj::Array<jsg::Ref<api::WebSocket>> getWebSockets(jsg::Lock& js, kj::String tag);
+  // TODO(now): More issues with StringPtr here?
+  //
+  // Gets an array of accepted WebSockets matching the given tag.
+  // Disconnected WebSockets are automatically removed from the list.
+
+  void setEventTimeout(size_t ms);
```

Comment:
What is this method for?

Reply:
It's part of the "Hibernation API" doc Kenton wrote, it prevents any event from running for too long (but won't be implemented in this PR). I think @jqmmes is working on this actually so I can probably remove it from my PR (it's a remnant of a previous PR where I was doing the JS side only).

Reply:
Yes, this is something I'll work on. Fell free to remove the method from your PR.

Reply:
https://github.com/cloudflare/workerd/pull/339/commits/0bb73cbeb8ae9c76419fafa9823d08c403c834ce